gpu/containers/src/device_memory.cpp

```diff
@@ -161,34 +161,6 @@ pcl::gpu::DeviceMemory2D::empty() const
 
 #else
 
-//////////////////////////    XADD    ///////////////////////////////
```

Comment:
Don't delete. Deprecate instead.

How to deprecate? Use std::atomic in CV_XADD function, deprecate the function  and remove everything else

Reply:
@larshg What's your view on this? Delete or deprecate?

Reply:
I'm fine with both. But I guess the safer approach is to deprecate like is done now. However, its probably not used elsewhere.

Reply:
I would generally also prefer the safer approach (deprecating), but in this case, I simply don't see how anyone else could possibly use `CV_XADD`: it is a macro/inline function in a cpp file, so unless someone `#includes` device_memory.cpp in their own code (obviously wrong), there is no way to have access to this?

Reply:
Yeah, that's true. I'm also fine with removing it.